### DISCUSSION TITLE
CORS error when login

### DISCUSSION THREAD

**Discussion by megaezz** [orchidsoftware/platform#2149] [Q&A]
I'm using Cloudflare Flexible. So I configured AppServiceProvider.php like this:
```
if ($this->app->environment() == 'production') {
            URL::forceScheme('https');
}
```
I open site with https, screens and forms works fine, except login.
When I press "Login" I get errors in Console:
```
[Error] Origin https://domain.com is not allowed by Access-Control-Allow-Origin.
[Error] Fetch API cannot load http://domain.com/admin due to access control checks.
[Error] Failed to load resource: Origin https://domain.com is not allowed by Access-Control-Allow-Origin. (admin, line 0)
[Error] Unhandled Promise Rejection: TypeError: Load failed
	(anonymous function)
	rejectPromise
	promiseReactionJob
```

And after page reloading - I am already logged in and everything is fine.

I tried to add '*' to config/cors.php
`'paths' => ['api/*', 'sanctum/csrf-cookie','*'],`
After this I get other errors in Console but same behaviour.
```
[Warning] [blocked] The page at https://domain.com/admin/login was not allowed to display insecure content from http://domain.com/admin.

[Error] Unhandled Promise Rejection: TypeError: Load failed
	(anonymous function)
	rejectPromise
	promiseReactionJob
```

So why platform trying to send request to 'http' when log in? How to fix it?

**Comment by tabuna** [comment]
See the following discussions https://github.com/orchidsoftware/platform/discussions/1671 and https://github.com/orchidsoftware/platform/discussions/1753

**Reply by megaezz** [reply]
Thanks a lot. #1753 - solution
